modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/KubernetesRuntimeProvisioningService.java

```diff
@@ -342,20 +342,207 @@ public DeploymentLogs getRuntimeLogs(DeploymentConfig deploymentConfig, Query qu
         return deploymentLogs;
     }
 
+    /**
+     * Set runtime properties to kubernetes environment
+     *
+     * @param runtimeProperties runtime properties
+     * @param deploymentConfig  includes deployment related details
+     * @throws RuntimeProvisioningException
+     */
     @Override
-    public void setRuntimeProperties(List<RuntimeProperty> runtimeProperties)
+    public void setRuntimeProperties(List<RuntimeProperty> runtimeProperties, DeploymentConfig deploymentConfig)
             throws RuntimeProvisioningException {
 
+        List secrets = new ArrayList();
+        Container container = new Container();
+        List environments = new ArrayList();
+
+        //create a instance of kubernetes client to invoke service call
+        KubernetesClient kubernetesClient = KubernetesProvisioningUtils.getFabric8KubernetesClient();
+
+        for (RuntimeProperty runtimeProperty : runtimeProperties) {
+            switch (runtimeProperty.getPropertyType()) {
+            case SENSITIVE:
+                if (log.isDebugEnabled()) {
+                    String message = "Creating property type secret for the application:" + applicationContext.getName()
+                            + " for the tenant domain:" + applicationContext.getTenantInfo().getTenantDomain();
+                    log.debug(message);
+                }
+                Secret secret = new SecretBuilder().withKind(KubernetesPovisioningConstants.KIND_SECRETS)
+                        .withApiVersion(Secret.ApiVersion.V_1).withNewMetadata()
```

Comment:
We do don't need specify the API version in each K8s kinds. Because we specify it on the connection creation time.
https://github.com/wso2/product-af/pull/800/commits/8b81ecdbd5250b7c8dfc60670c6d8f07283be906#diff-a7bada20489a1c9b18bde9085a12b5daR59